app/Jobs/RockTheVote/ImportRockTheVoteRecord.php

```diff
@@ -258,26 +260,101 @@ public function updateUserIfChanged(NorthstarUser $user)
+        $payload = array_merge($payload, $this->parseSmsStatusChangeForUser($user));
+
+        return $payload;
+    }
+
+    /**
+     * Returns payload to update mobile if user does not currently have one saved.
+     *
+     * @return array
+     */
+    public function parseMobileChangeForUser(NorthstarUser $user)
+    {
+        $fieldName = 'mobile';
+
+        if ($user->{$fieldName}) {
+            return [];
         }
 
-        // @TODO: Check for changes to user's SMS status and subscription topics.
+        return [$fieldName => $this->userData[$fieldName]];
+    }
+
+    /**
+     * Returns payload to update SMS subscription topics if they have changed.
+     *
+     * @return array
+     */
+    public function parseSmsSubscriptionTopicsChangeForUser(NorthstarUser $user)
+    {
+        $fieldName = 'sms_subscription_topics';
+        $currentSmsTopics = ! empty($user->{$fieldName}) ? $user->{$fieldName} : [];
+
+        // If user opted in to SMS, add the import topics to current topics.
+        if ($this->smsOptIn) {
+            return [$fieldName => array_unique(array_merge($currentSmsTopics, $this->userData[$fieldName]))];
+        }
+
+        // Nothing to remove if current topics in empty.
+        if (! count($currentSmsTopics)) {
+            return [];
+        }
+
+        // If user hasn't opted-in and has current topics, remove all import topics from current.
+        $updatedSmsTopics = [];
+
+        foreach ($currentSmsTopics as $topic) {
+            if (! in_array($topic, explode(',', config('import.rock_the_vote.user.sms_subscription_topics')))) {
+                array_push($updatedSmsTopics, $topic);
+            }
+        }
+
+        return [$fieldName => $updatedSmsTopics];
+    }
+
+    /**
+     * Returns payload to update SMS status if it has changed.
+     *
+     * @return array
+     */
+    public function parseSmsStatusChangeForUser(NorthstarUser $user)
+    {
+        $fieldName = 'sms_status';
+        $currentSmsStatus = $user->{$fieldName};
+        $importSmsStatus = $this->userData[$fieldName];
+
+        /**
+         * If current status is null or undeliverable, update status per whether they opted in
+         * via the RTV form.
+         *
+         * This is the only scenario when we want to change an existing user's status to stop.
+         */
+        if ($currentSmsStatus == SmsStatus::$undeliverable || ! $currentSmsStatus) {
+            return [$fieldName => $importSmsStatus];
+        }
 
-        // Update user's mobile only if we currently don't have one saved.
-        if (! $user->mobile) {
-            $result['mobile'] = $this->userData['mobile'];
+        if ($this->smsOptIn && in_array($currentSmsStatus, [
+            SmsStatus::$less,
+            SmsStatus::$pending,
+            SmsStatus::$stop,
+        ])) {
+            return [$fieldName => $importSmsStatus];
         }
 
-        return $result;
+        return [];
```

Comment:
Is this something we could handle in Northstar – if we add a topic for someone and they're set to `STOP`, we could automatically set their status to `active`? (I believe  that's how we handle changes to email topics, right?)

Reply:
This touches on https://github.com/DoSomething/northstar/pull/1008 -- subscribing to SMS has the variation that you could be re-subscribing with a `less` or `active` status. The email subscription topic is easier to automate since the email subscription status is boolean.